### PR TITLE
update remote-data-services after fsspec [skip ci]

### PR DESCRIPTION
**Pull request by martindurant**
- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: docs/source/remote-data-services.rst [hunk 1]
```diff
@@ -32,13 +35,17 @@ codebase:
 - **HTTP(s)**: ``http://`` or ``https://`` for reading data directly from HTTP web servers
 
 - **Azure Datalake Storage**: ``adl://``, for use with the Microsoft
-  Azure platform, using azure-data-lake-store-python_. This is *experimental*, use at your own risk.
+  Azure platform, using azure-data-lake-store-python_, is unavailable in the current release of
+  ``fsspec``, but a new version using Microsoft's "protocol 2" should come soon.
+
+``fsspec`` also provides other file sytstems that may be of interest to Dask users, such as
+ssh, ftp, webhdfs - see the documentation.
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]:
```suggestion
ssh, ftp, and webhdfs.  See the documentation for more information.
```

File: docs/source/remote-data-services.rst [hunk 2]
```diff
@@ -15,13 +15,16 @@ functions like ``dd.read_csv``:
...
+as loading of configuration, is done using ordinary python method.
+
 The following remote services are well supported and tested against the main
 codebase:
 
 - **Local or Network File System**: ``file://`` - the local file system, default in the absence of any protocol
 
 - **Hadoop File System**: ``hdfs://`` - Hadoop Distributed File System, for resilient, replicated
-  files within a cluster. Can use either hdfs3_ or pyarrow_.
+  files within a cluster. Uses pyarrow_ as the backend.
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]: Just to make this a complete sentence

```suggestion
  files within a cluster. This uses pyarrow_ as the backend.
```

File: docs/source/remote-data-services.rst [hunk 3]
```diff
@@ -52,7 +59,10 @@ Optional Parameters
 
 Two methods exist for passing parameters to the backend file system driver:
 extending the URL to include username, password, server, port, etc.; and
-providing ``storage_options``, a dictionary of parameters to pass on. Examples:
+providing ``storage_options``, a dictionary of parameters to pass on. The
+second form is more general, as any number of file system-specific options
+can be passed.
+Examples:
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]: Did you intend for `Examples` to be on a new line?

File: docs/source/remote-data-services.rst [hunk 4]
```diff
@@ -61,16 +71,22 @@ providing ``storage_options``, a dictionary of parameters to pass on. Examples:
    df = dd.read_parquet('s3://bucket/path',
                         storage_options={'anon': True, 'use_ssl': False})
 
-Further details on how to provide configuration for each back-end
-is listed next.
+Further details on how to provide configuration for the main back-ends
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]: We say further details twice here

```suggestion 
Details on how to provide configuration for the main back-ends
```

File: docs/source/remote-data-services.rst [hunk 5]
```diff
@@ -61,16 +71,22 @@ providing ``storage_options``, a dictionary of parameters to pass on. Examples:
...
+is listed next, but further details can be found in the docs pages of the
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]:
```suggestion
is listed next, but further details can be found in the documentation of the
```

File: docs/source/remote-data-services.rst [hunk 6]
```diff
@@ -61,16 +71,22 @@ providing ``storage_options``, a dictionary of parameters to pass on. Examples:
...
+relavant back-end.
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]:
```suggestion
relevant back-end.
```

We should probably run spellcheck on the file as a whole

File: docs/source/remote-data-services.rst [hunk 7]
```diff
@@ -61,16 +71,22 @@ providing ``storage_options``, a dictionary of parameters to pass on. Examples:
...
 
 Each back-end has additional installation requirements and may not be available
-at runtime. The dictionary ``dask.bytes.core._filesystems`` contains the
-currently available file systems. Some require appropriate imports before use.
+at runtime. The dictionary ``fsspec.registry`` contains the
+currently imported file systems, and to see which backends ``fsspec`` knows how
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]:
```suggestion
currently imported file systems.  To see which backends ``fsspec`` knows how
```

File: docs/source/remote-data-services.rst [hunk 8]
```diff
@@ -61,16 +71,22 @@ providing ``storage_options``, a dictionary of parameters to pass on. Examples:
...
 
-The following list gives the protocol shorthands and the back-ends to which
-they refer.
+.. code-block:: python
 
+    from fsspec.registry import known_implementations
+    known_implementations
+
+Note that some backends appear twice, if they can be referenced with multiple
+protocol strings, e.g., "http" and "https".
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]: Copy-editing people often yell at me for using "e.g.", which is apparently unclear for non-native english speakers

```suggestion
protocol strings, like "http" and "https".
```

Honestly, even as a native english speaker I'm not entirely sure what it means :)

File: docs/source/remote-data-services.rst [hunk 9]
```diff
@@ -95,49 +111,14 @@ Hadoop File System
...
-    dd.read_csv('hdfs:///path/to/*.csv')
-
-   # Use hdfs3 for HDFS I/O
-   with dask.config.set(hdfs_driver='hdfs3'):
-       dd.read_csv('hdfs:///path/to/*.csv')
-
-   # Use pyarrow for HDFS I/O
-   with dask.config.set(hdfs_driver='pyarrow'):
-       dd.read_csv('hdfs:///path/to/*.csv')
+Spark. HDFS support can be provided by pyarrow_.
```
Comment (mrocklin) [inline docs/source/remote-data-services.rst]:
```suggestion
Spark. HDFS support can be provided by PyArrow_.
```

Given that we're capitalizing Hadoop, Spark, and HDFS, we should probably capitalize PyArrow as well to show equal respect.

**Comment by martindurant** [comment]
Suggest this is done.

**Pull request merged**